ember.s

```diff
@@ -147,9 +147,42 @@ not_poke:
 
 	call verify_end
 	call di
-	jmp short shell
+	jmp shell
 
 not_run:
+	cmp al, 'w'
+	jnz short .not_write
+	mov cl, 0x00
+	jmp short .continue
+.not_write:
+	cmp al, 'r'
+	mov cl, 0x01
+	jnz short not_rw
+.continue:	
+        mov bx, di ; move the read location
+	call readhexword
+        jc short parse_error
+	shl eax, 16
+	call readhexword
+        jc short parse_error
+	; eax should hold our lba now
+
+	mov di, bx
+	push es
+	push cs
+	pop es
```

Comment:
CS=DS=ES=SS=0 is already initialized on boot.

Reply:
yeah but what if you call that interrupt with different segment registers?

Reply:
This particular code is part of the command dispatch, which doesn't run in interrupts, no?